### PR TITLE
update blueline, bonito, coral, crosshatch, flame, sargo, sunfish for android 11

### PR DESCRIPTION
**Pull request by dan-v**

### CODE REVIEW

File: coral/config.json [hunk 1]
```diff
@@ -26,321 +26,200 @@
   "api-30": {
     "naked": {
       "overlays-dir": "",
-      "product-bytecode": [
-        "product/app/QtiTelephonyService/QtiTelephonyService.apk",
-        "product/app/datastatusnotification/datastatusnotification.apk",
-        "product/app/uceShimService/uceShimService.apk",
-        "product/priv-app/ims/ims.apk",
```
Comment (chirayudesai) [inline coral/config.json:33]: This is now in system_ext

https://git.rip/dumps/google/flame/-/blob/flame-user-11-RP1A.200720.009-6720564-release-keys/all_files.txt

File: flame/config.json [hunk 2]
```diff
@@ -26,321 +26,200 @@
...
-        "product/lib64/com.quicinc.cne.api@1.1.so",
-        "product/lib64/com.quicinc.cne.constants@1.0.so",
-        "product/lib64/com.quicinc.cne.constants@2.0.so",
-        "product/lib64/com.quicinc.cne.constants@2.1.so",
-        "product/lib64/lib-imsvideocodec.so",
-        "product/lib64/lib-imsvt.so",
-        "product/lib64/lib-imsvtextutils.so",
-        "product/lib64/lib-imsvtutils.so",
-        "product/lib64/libaptXHD_encoder.so",
-        "product/lib64/libaptX_encoder.so",
```
Comment (chirayudesai) [inline flame/config.json:74]: These are also in system_ext now.

File: flame/config.json [hunk 3]
```diff
@@ -26,321 +26,200 @@
...
         "bin/hw/android.hardware.nfc@1.2-service.st",
-        "bin/hw/android.hardware.power@1.3-service.pixel-libperfmgr",
+        "bin/hw/android.hardware.power-service.pixel-libperfmgr",
         "bin/hw/android.hardware.power.stats@1.0-service.pixel",
         "bin/hw/android.hardware.secure_element@1.0-service.st",
+        "bin/hw/android.hardware.sensors@2.0-service.multihal",
         "bin/hw/android.hardware.thermal@2.0-service.pixel",
         "bin/hw/android.hardware.usb@1.2-service.coral",
-        "bin/hw/android.hardware.vibrator@1.3-service.coral",
+        "bin/hw/android.hardware.vibrator-service.cs40l25",
```
Comment (chirayudesai) [inline flame/config.json:214]: I had to do a double take on this one

**Review by chirayudesai — CHANGES_REQUESTED** [review]
The good: A lot of removals, and not as many modifications / new additions. Especially for 3, 3a families.
4 and 4a have more changes.

The bad: Some of those removals are actually just moves to `system_ext` which we don't support right now.

It'd be trivial to add it in based on the code we added last year for `product`, but do we want to at this point?

**Comment by chirayudesai** [comment]
Dan, would you mind just diffing the config.json for
- crosshatch and blueline
- bonito and sargo
- coral and flame
- coral and sunfish

And making sure they're sycned up  / the diff looks sane. For the first three I don't think there really would be any changes.

**Comment by chirayudesai** [comment]
Note: I didn't exhaustively go through the full thing since the list is a lot bigger, but in general since it's most removals / additions to vendor-skip-files, as long as everything works we should be good.

**Comment by danielfullmer** [comment]
There are also some differences between the built `vendor/build.prop` and Google's `vendor/build.prop` on `crosshatch` (and likely others):
```diff
diff --git a/upstream/vendor/build.prop b/built/vendor/build.prop
index 9a767f3..5675f99 100644
--- a/upstream/vendor/build.prop
+++ b/built/vendor/build.prop
.....
 #
 # ADDITIONAL VENDOR BUILD PROPERTIES
 #
 ro.build.expect.bootloader=b1c1-0.3-6623201
 ro.build.expect.baseband=g845-00107-200702-B-6648703
+ro.carrier=unknown
+ro.config.notification_sound=OnTheHunt.ogg
+ro.config.alarm_alert=Alarm_Classic.ogg
 ro.apex.updatable=true
 keyguard.no_require_sim=true
 ro.control_privapp_permissions=enforce
@@ -138,15 +141,3 @@ aaudio.hw_burst_min_usec=2000
 ro.bluetooth.a2dp_offload.supported=true
 persist.bluetooth.a2dp_offload.disabled=false
 persist.bluetooth.a2dp_offload.cap=sbc-aac-aptx-aptxhd-ldac
-ro.hardware.vulkan=adreno
-ro.hardware.egl=adreno
-vendor.power.pasr.enabled=true
-ro.boot.vendor.overlay.theme=com.android.internal.systemui.navbar.twobutton
-drm.service.enabled=true
-media.mediadrmservice.enable=true
-ro.com.android.prov_mobiledata=false
-ro.storage_manager.enabled=true
-ro.carrier=unknown
-ro.config.notification_sound=OnTheHunt.ogg
-ro.config.alarm_alert=Alarm_Classic.ogg
```

At the very least, we should probably set `ro.hardware.vulkan` and `ro.hardware.egl` in `PRODUCT_PROPERTY_OVERRIDES` like is done in taimen/walleye

**Comment by dan-v** [comment]
I took another spin through these devices and added system_ext. I didn't quite get a chance to get to sunfish tonight.
* diff between device pairs is now just the `lib/hw/sensors.<device>.so` and `lib64/hw/sensors.<device>.so`.
* the system_ext diffs for pixel 4/4a devices are quite large. i picked up most changes and figured we could pair down.
* i've only validated that crosshatch config boots as it previously did
* i'm kicking off builds for all of these devices

**Comment by thestinger** [comment]
Need to deal with the system property differences that are specific to devices but this is a good start.

**Comment by thestinger** [comment]
> drm.service.enabled=true

This controls whether DRM is supported. It should be up to the OS.

> media.mediadrmservice.enable=true

This controls whether proper secure buffers are supported so that the sandbox can be used. This should probably be enabled by android-prepare-vendor, rather than the OS. It doesn't enable DRM but rather sandboxing it securely if it's enabled.

**Comment by thestinger** [comment]
Going to file a separate issue about system properties.

**Comment by thestinger** [comment]
https://github.com/AOSPAlliance/android-prepare-vendor/issues/27

**Comment by thestinger** [comment]
Going to merge this since it's a lot better than the current state. Can handle any issues as follow-up.

**Pull request merged**